lib/matrix_convert_to_zero.rb

```diff
@@ -2,6 +2,79 @@
 # Assumption/ Given: All numbers in the matrix are 0s or 1s
 # If any number is found to be 0, the method updates all the numbers in the
 # corresponding row as well as the corresponding column to be 0.
+
+# matrix[row, column]
+require 'pry'
+
 def matrix_convert_to_0(matrix)
-  raise NotImplementedError
+
+  # attempted to avoid using matrix.size.  not functional.
+  # i = 0
+  # rows = 0
+  #
+  # until matrix[i][0] == nil #get number of rows
+  #   rows += 1
+  #   i += 1
+  # end
+  #
+  # j = 0
+  # columns = 0
+  #
+  # until matrix[0, j] == nil #get number of columns
+  #   columns += 1
+  #   j += 1
+  # end
+
+
+#time complexity O(n^2), space complexity O(n + m) where
+#n and m are row and column sizes
+
+  rows = matrix.size
+  columns = matrix[0].size
+
+  rows_to_zero = []
+  columns_to_zero = []
+
+  i = 0
+  until i == rows
+
+    j = 0
+    until j == columns
+
+      if matrix[i][j] == 0
+        rows_to_zero << i
+        columns_to_zero << j
+      end
+
+      j += 1
+
+    end
+    i += 1
+
```

Comment:
Something seems wrong: on line 52, you increment `i` by one and then on line 54, you make the value of i 0 again.